pytorch_lightning/trainer/trainer.py

```diff
@@ -135,6 +135,7 @@ def __init__(
         move_metrics_to_cpu: bool = False,
         enable_pl_optimizer: bool = None,  # todo: remove in v1.3
         multiple_trainloader_mode: str = 'max_size_cycle',
+        deepspeed_config: Optional[Union[Path, str, Dict]] = None,
```

Comment:
personally, I would think about adding `**kwargs` where we can have all these new/experimental arguments

Reply:
it would also give us more freedom ale pain with eventual deprecation...

Reply:
how should we define it though? like accelerator kwargs?

We had the discussion internally, let me bring it up back up! I personally think we should expose it via the config for something I know will be useful, but the functionality works either way0

Reply:
yes, my point is not to introduce a new argument, just hide it in `kwargs` so
the trainer definition would be 
```python
class Trainer(...):
  def __init__(self, ..., **kwargs): # the .. stand for all existing arguments
    ...
    self.deepspeed_config = kwargs.get('deepspeed_config')
    ...
```
eventually, we can add an extra check that kwargs does not include any typos-leftovers
so for user he can do the same as with the args
```python
t = Trainer(..., accelerator='deepspeed', deepspeed_config='path/to/a/file')
```

Reply:
it is a similar case as the `find_unused_parameters` config for DDP.
@Borda minor suggestion for improvement in case you decide to go with it: 
we should do `kwargs.pop`
and at the end we just make sure kwargs is empty. We should throw errors if an unknown kwarg is passed in, otherwise typo errors in argument names are silently ignored.

Reply:
Had the discussion offline we're going for:

```python
    model = BoringModel()
    trainer = Trainer(
        plugins=[DeepSpeedPlugin(deepspeed_config)],
        gpus=1,
        fast_dev_run=True,
    )
```
or
```bash
DEEPSPEED_CONFIG_PATH=ds.json python train.py accelerator='deepspeed' gpus=1
```